ships/0035-beta-api-changes.md

```diff
@@ -0,0 +1,119 @@
+-->
+
+---
+title: Shipwright BETA API Release
+authors:
+  - "@qu1queee"
+reviewers:
+  - "@SaschaSchwarze0"
+  - "@coreydaley"
+  - "@otaviof"
+  - "@dalbar"
+  - "@adambkaplan"
+approvers:
+  - "@SaschaSchwarze0"
+  - "@adambkaplan"
+creation-date: 2022-07-09
+last-updated: 2022-07-09
+status: implementable
+---
+
+# Shipwright BETA API Release
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Open Questions [optional]
+
+## Summary
+
+Propose a path to release Shipwright Beta API. Achieving a Beta API will provide users a more stable version they can depend on, with the certainty of not dropping features. In this document we provide the modifications, deprecations or additions to the Beta API.
+
+## Motivation
+
+1. The majority of the Shipwright Alpha API did not change over time. Part of the current Alpha API is considered mature enough for graduating to Beta.
+
+2. The Shipwright API got many novel ideas over time, supporting different use cases. These features were implemented as part of the Alpha API, but the desired state was not reach. Planning for Beta API puts into triage these features and their potential deprecation.
+
+3. The Shipwright API was created with some assumptions in mind, over time new features force us to rethink on better states of the API.
+
+### Goals
+
+1. Layout the changes for Beta API.
+
+2. Document the deprecation policy for Beta API.
+
+3. Define supported use cases for Beta API.
+
+### Non-Goals
+
+1. Document migration strategies of API conventions or object definitions from Alpha to Beta.
+
+## Proposal
+
+We follow the Kubernetes API [versioning](https://kubernetes.io/docs/reference/using-api/#api-versioning). As documented by Kubernetes, we stick to the same levels (_Alpha, Beta or Stable_), depending on the levels of stability and support.
+
+### Deprecation Policy
+
+For the policy of Beta, we rely also on the Kubernetes Deprecation [policies](https://kubernetes.io/docs/reference/using-api/deprecation-policy/), which translates to the following four rules:
+
+1. _API elements may only be removed by incrementing the version of the API group._
+2. _API objects must be able to round-trip between API versions in a given release without information loss, with the exception of whole REST resources that do not exist in some versions._
+3. _An API version in a given track may not be deprecated in favor of a less stable API version._
+4. _Minimum API lifetime is determined by the API stability level_
+
+Related to 4), in this document we propose that for Beta API versions, we provide coverage after deprecation for **9** months. We do not encourage to be based on releases quantity, per the faster cadence we have compared to Kubernetes.
```

Comment:
This to me is a bit unclear - after reading the deprecation policy doc, I think we can make this a bit more succinct. Hopefully I am capturing your intent here:

```suggestion
In the spirit of the Kubernetes Deprecation [policies](https://kubernetes.io/docs/reference/using-api/deprecation-policy/), the beta API will come with the following support guarantees:

1. No breaking changes, such as removal of fields, will be introduced without introducing a new beta API version (ex - `v1beta2`).
2. Beta API versions may be deprecated, however they will continue to be served and supported for a period of 9 months after the deprecation is announced.
3. The beta API will never be deprecated in favor of an `alpha` API (ex - `v1alpha2`).
```

Reply:
Upstream guidance also suggests that we don't have to create a converting webhook and keep serving alpha APIs. I think we as a community won't do that because we don't want the current user base to lift and shift their Shipwright data to new clusters. I think we can shorten the support duration of the alpha APIs to less than 9 months - perhaps 2-3 releases?

Reply:
Thanks, this is more digestible, I implemented the changes. I kept the 9 months, we can shorten it. 2-3 releases sound ideal, as long as we have a good understanding of our current cadence :)